### DISCUSSION TITLE
Export Links / Backup

### DISCUSSION THREAD

**Discussion by datoma** [karakeep-app/karakeep#396] [Ideas]
Hi, is there a function to backup all Links and stuff stored in hoarder or do I have to backup the sql-file?
Would it be possible to get a function "export to json/txt/whatever" so I can have the links tags and maybe a summary of the link in a file?

**Comment by kamtschatka** [comment]
Yes, you backup the sqlite database.
You can already export it as json by using the API and providing the --json option: 
`--api-key <api-key> --server-addr <server-address> --json bookmarks list`

**Reply by raviwarrier** [reply]
Hi. I incorrectly installed Hoarder in the wrong folder on my raspberry pi, imported some 2000 bookmarks and then realized my mistake. I am noob when it comes to calling API functions and such, so, I would deeply appreciate if you could tell me how to call --api-key ... 

I use a windows machine to access hoarder through the webui, if that helps.

**Reply by kamtschatka** [reply]
sounds like you would export everything and import again? If that is the case I would rather just move the database file to the new location and that would be it?

**Reply by raviwarrier** [reply]
I've been trying to figure this out but I'm getting confused about where to place everything. I’ve already tried about 4 times but ended up reverting back to using the original container because it didn’t work as expected.

Here’s the situation:
In my var/lib/... directory, I have two folders: hoarder_data and hoarder_meilisearch, and both of them contain a _data subfolder.

I’m trying to install the app on my SSD at /srv/ssd/AppData/local/hoarder-app, which is what I'll use in the volumes: section of my docker-compose file.

So my question is:

    Should I copy the entire hoarder_data and hoarder_meilisearch folders into /local/hoarder-app?
    Or should I only copy the _data subfolders inside them?
    Alternatively, do I need to manually create data and meilisearch folders under /local/hoarder-app and then move the _data subfolders into those?

Please let me know if anything is unclear or if more details would help.

The reason why this is confusing is because if I do the default install using the sample docker compose, it creates the two _data and _meilisearch folders. but if I put /srv/ssd/Appdata/local/hoarder-app:data in the compose file, it doesn't...

I am not a techie and still not comfortable with self-hosting, so, please be kind to ignore my naivety.
